castai/resource_workload_scaling_policy.go

```diff
@@ -0,0 +1,341 @@
+		UpdateContext: resourceWorkloadScalingPolicyUpdate,
+		DeleteContext: resourceWorkloadScalingPolicyDelete,
+		CustomizeDiff: resourceWorkloadScalingPolicyDiff,
+		Importer: &schema.ResourceImporter{
+			StateContext: workloadScalingPolicyImporter,
+		},
+		Description: "Manage workload scaling policy. Scaling policy [reference](https://docs.cast.ai/docs/woop-scaling-policies)",
+		Schema: map[string]*schema.Schema{
+			FieldClusterID: {
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				Description:      "CAST AI cluster id",
+				ValidateDiagFunc: validation.ToDiagFunc(validation.IsUUID),
+			},
+			"name": {
+				Type:             schema.TypeString,
+				Required:         true,
+				Description:      "Scaling policy name",
+				ValidateDiagFunc: validation.ToDiagFunc(validation.StringMatch(k8sNameRegex, "name must adhere to the format guidelines of Kubernetes labels/annotations")),
+			},
+			"apply_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				Description: `Recommendation apply type. 
+	- IMMEDIATE - pods are restarted immediately when new recommendation is generated.
+	- DEFERRED - pods are not restarted and recommendation values are applied during natural restarts only (new deployment, etc.)`,
+				ValidateDiagFunc: validation.ToDiagFunc(validation.StringInSlice([]string{"IMMEDIATE", "DEFERRED"}, false)),
+			},
+			"management_option": {
+				Type:     schema.TypeString,
+				Required: true,
+				Description: `Defines possible options for workload management.
+	- READ_ONLY - workload watched (metrics collected), but no actions performed by CAST AI.
+	- MANAGED - workload watched (metrics collected), CAST AI may perform actions on the workload.`,
+				ValidateDiagFunc: validation.ToDiagFunc(validation.StringInSlice([]string{"READ_ONLY", "MANAGED"}, false)),
+			},
+			"cpu": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem:     resourceSchema("QUANTILE", 0),
+			},
+			"memory": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem:     resourceSchema("MAX", 0.1),
+			},
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(15 * time.Second),
+			Read:   schema.DefaultTimeout(15 * time.Second),
+			Update: schema.DefaultTimeout(15 * time.Second),
+			Delete: schema.DefaultTimeout(15 * time.Second),
+		},
+	}
+}
+
+func resourceSchema(function string, overhead float64) *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"function": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				Description:      "The function used to calculate the resource recommendation. Supported values: `QUANTILE`, `MAX`",
+				Default:          function,
+				ValidateDiagFunc: validation.ToDiagFunc(validation.StringInSlice([]string{"QUANTILE", "MAX"}, false)),
+			},
+			"args": {
+				Type:     schema.TypeList,
```

Comment:
maybe stupid question, but why is this not TypeFloat when it should be a single float number?

Reply:
I guess we had idea to support multiple args initially